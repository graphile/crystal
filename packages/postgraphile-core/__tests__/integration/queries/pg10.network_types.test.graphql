query {
  inetTest: allNetworks(condition: { inet: "192.168.0.0" }) { ...networkConnection }
  cidrTest: allNetworks(condition: { cidr: "192.168.0.0/16" }) { ...networkConnection }
  macaddrTest: allNetworks(condition: { macaddr: "08:00:2b:01:02:03" }) { ...networkConnection }
  macaddr8Test: allNetworks(condition: { macaddr8: "08:00:2b:01:02:03:04:05" }) { ...networkConnection }
}

fragment networkConnection on NetworksConnection {
  pageInfo {
    startCursor
    endCursor
    hasNextPage
    hasPreviousPage
  }
  totalCount
  edges {
    cursor
    node {
      id
      inet
      cidr
      macaddr
      macaddr8
    }
  }
}
