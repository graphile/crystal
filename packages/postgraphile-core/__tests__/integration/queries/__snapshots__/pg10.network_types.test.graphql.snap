// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute 1`] = `
Object {
  "data": Object {
    "cidrTest": Object {
      "edges": Array [
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
          "node": Object {
            "cidr": "192.168.0.0/16",
            "id": 1,
            "inet": "192.168.0.0",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
      },
      "totalCount": 1,
    },
    "inetTest": Object {
      "edges": Array [
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
          "node": Object {
            "cidr": "192.168.0.0/16",
            "id": 1,
            "inet": "192.168.0.0",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
      },
      "totalCount": 1,
    },
    "macaddr8Test": Object {
      "edges": Array [
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
          "node": Object {
            "cidr": "192.168.0.0/16",
            "id": 1,
            "inet": "192.168.0.0",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=",
          "node": Object {
            "cidr": "192.168.0.0/24",
            "id": 2,
            "inet": "192.168.0.1",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFszXV0=",
          "node": Object {
            "cidr": "172.16.0.0/12",
            "id": 3,
            "inet": "172.16.0.0/12",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
          "node": Object {
            "cidr": "172.16.0.0/24",
            "id": 4,
            "inet": "172.16.0.1/12",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs1XV0=",
          "node": Object {
            "cidr": "2001:4f8:3:ba::/64",
            "id": 5,
            "inet": "2001:4f8:3:ba::",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs2XV0=",
          "node": Object {
            "cidr": "2001:4f8:3:ba:2e0:81ff:fe22:d1f1/128",
            "id": 6,
            "inet": "2001:4f8:3:ba:2e0:81ff:fe22:d1f1",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs3XV0=",
          "node": Object {
            "cidr": "::ffff:1.2.3.0/120",
            "id": 7,
            "inet": "::ffff:1.2.3.0/120",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs4XV0=",
          "node": Object {
            "cidr": "::ffff:1.2.3.0/128",
            "id": 8,
            "inet": "::ffff:1.2.3.0",
            "macaddr": "aa:bb:cc:dd:ee:ff",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs4XV0=",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
      },
      "totalCount": 8,
    },
    "macaddrTest": Object {
      "edges": Array [
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
          "node": Object {
            "cidr": "192.168.0.0/16",
            "id": 1,
            "inet": "192.168.0.0",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=",
          "node": Object {
            "cidr": "192.168.0.0/24",
            "id": 2,
            "inet": "192.168.0.1",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFszXV0=",
          "node": Object {
            "cidr": "172.16.0.0/12",
            "id": 3,
            "inet": "172.16.0.0/12",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
          "node": Object {
            "cidr": "172.16.0.0/24",
            "id": 4,
            "inet": "172.16.0.1/12",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs1XV0=",
          "node": Object {
            "cidr": "2001:4f8:3:ba::/64",
            "id": 5,
            "inet": "2001:4f8:3:ba::",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs2XV0=",
          "node": Object {
            "cidr": "2001:4f8:3:ba:2e0:81ff:fe22:d1f1/128",
            "id": 6,
            "inet": "2001:4f8:3:ba:2e0:81ff:fe22:d1f1",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
        Object {
          "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs3XV0=",
          "node": Object {
            "cidr": "::ffff:1.2.3.0/120",
            "id": 7,
            "inet": "::ffff:1.2.3.0/120",
            "macaddr": "08:00:2b:01:02:03",
            "macaddr8": "08:00:2b:01:02:03:04:05",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs3XV0=",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
      },
      "totalCount": 7,
    },
  },
}
`;
